io/disk/dbench.py

```diff
@@ -25,12 +26,9 @@ def setUp(self):
         Source:
         http://samba.org/ftp/tridge/dbench/dbench-3.04.tar.gz
         '''
-
-        try:
-            process.run('which gcc')
-        except:
-            self.error('gcc is required by this job and is not available on'
-                       'the system')
+        sm = SoftwareManager()
+        if not sm.check_installed("gcc"):
+            sm.install("gcc")
```

Comment:
`install()` returns `False` on install failure. You can just:
~~~
...
        if not sm.check_installed("gcc") and not sm.install("gcc"):
            self.error('need gcc')
...
~~~